### PR TITLE
[9.0][database_cleanup] Fix models purging

### PR DESCRIPTION
**Pull request by florian-dacosta**
PR relative to this issue : 
https://github.com/OCA/server-tools/issues/587

### CODE REVIEW

File: database_cleanup/models/ir_model_fields.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    # installed module. In this specific case, we need to avoid to check
+    # if fields can be removed as it would fail.
+    @api.multi
+    def _prepare_update(self):
+        if self.env.context.get('purge'):
+            not_available_fields = self.browse(False)
+            for record in self:
+                try:
+                    self.env[record.model]
+                except:
```
Comment (hbrunn) [inline database_cleanup/models/ir_model_fields.py]: you can avoid the exception handling (which is always dangerous to catch stuff unconditionally) by asking the registry embedded in `self.env.registry`, and condense everything to

``` python
self -= self.filtered(lambda x: x.model not in self.env.registry)
```

**Review by hbrunn — CHANGES_REQUESTED** [review]
also add a fixes #issuenumber to your commit message to have the issue closed on merging, and to give a future reference why we want this: https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by florian-dacosta** [comment]
Thank you for the advices, I've made the changes

**Review by hbrunn — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Review by alan196 — APPROVED** [review]
I just tested it!! Works perfectly

**Comment by pedrobaeza** [comment]
I have just merged the PR, but it has broken Travis CI. Can you please check?

**Comment by Garamotte** [comment]
@pedrobaeza The issue is that tests had been modified between the start point of this PR and the merge time (commit ab51acc).

I ran tests on the merged branch with the changes (I basically chery-picked ab51acc on e0ef553), and the test fails.

**Comment by pedrobaeza** [comment]
Well, the way Travis works is difficult to prevent this, unless we rebase before each merge.

**Pull request merged**